tests/settings_tests/tests.py

```diff
@@ -441,3 +443,119 @@ def test_tuple_settings(self):
+
+    Do this with three cases of failure:
+
+    1) receiver fails when the signal is sent while both entering and exiting the context manager,
+    2) receiver fails on enter,
+    3) receiver fails on exit.
+
+    Make sure that in each case ALL receivers are called when exiting the context manager.
+    """
+
+    def setUp(self):
+        signals.setting_changed.connect(self.receiver)
+
+        # Create a spy that will be connnected to `setting_changed` signal
+        # and executed AFTER `self.receiver`.
+        self.spy_receiver = mock.Mock()
+        signals.setting_changed.connect(self.spy_receiver)
+
+    def tearDown(self):
+        signals.setting_changed.disconnect(self.receiver)
+        signals.setting_changed.disconnect(self.spy_receiver)
+
+    def receiver(self, **kwargs):
+        """
+        A receiver for demonstration purposes.
+        It fails while certain settings are being changed.
+
+        - SECURE_BROWSER_XSS_FILTER raises an error while receiving the signal
+          on both entering and exiting the context manager
+        - SECURE_CONTENT_TYPE_NOSNIFF raises an error only on enter.
+        - SECURE_SSL_REDIRECT raises an error only on exit.
+        """
+        setting = kwargs['setting']
+        enter = kwargs['enter']
+
+        if setting in ('SECURE_BROWSER_XSS_FILTER', 'SECURE_CONTENT_TYPE_NOSNIFF') and enter:
+            raise SettingChangeEnterException
+        if setting in ('SECURE_BROWSER_XSS_FILTER', 'SECURE_SSL_REDIRECT') and not enter:
+            raise SettingChangeExitException
+
+    def check_settings(self):
+        """
+        Ensure that certain settings are still configured to their default values.
+        """
+        self.assertEqual(settings.SECURE_BROWSER_XSS_FILTER, global_settings.SECURE_BROWSER_XSS_FILTER)
+        self.assertEqual(settings.SECURE_CONTENT_TYPE_NOSNIFF, global_settings.SECURE_CONTENT_TYPE_NOSNIFF)
+        self.assertEqual(settings.SECURE_SSL_REDIRECT, global_settings.SECURE_SSL_REDIRECT)
+        self.assertEqual(settings.SECURE_HSTS_PRELOAD, global_settings.SECURE_HSTS_PRELOAD)
+
+    def check_spy_receiver_exit_calls(self, call_count):
+        """
+        Ensure that `self.spy_receiver` was called exactly `call_count` times
+        with ``enter=False`` keyword argument.
+
+        :type call_count: int
+        """
+        kwargs_with_exit = [kwargs for args, kwargs in self.spy_receiver.call_args_list
+                            if ('enter', False) in kwargs.items()]
+
+        self.assertEqual(len(kwargs_with_exit), call_count)
+
+    def test_override_settings_both(self):
+        """
+        Receiver fails on both enter and exit.
+        """
+        with self.assertRaises(SettingChangeEnterException):
+            with override_settings(SECURE_HSTS_PRELOAD=not global_settings.SECURE_HSTS_PRELOAD,
+                                   SECURE_BROWSER_XSS_FILTER=not global_settings.SECURE_BROWSER_XSS_FILTER):
```

Comment:
I'd use generic setting names to make it clear what's going to raise the exception and to avoid adjusting these settings if we ever get rid/deprecate one of these settings.

Reply:
@slafs non-existing setting names, e.g. `TEST_SETTING=True`

Reply:
Right, but how do I check its default value then (in the `check_settings` method here)? Or should I just check that those settings doesn't exist outside the context?

Reply:
> Or should I just check that those settings doesn't exist outside the context?

I think that'd work.

Reply:
Done in cfa613be5c72a7